### ISSUE TITLE
[bug] Failing CI for OPENSSL

### ISSUE THREAD

**Issue by micaelmalta** [dogecoin/dogecoin#2400]
# Bug Report

CI is failing because of openssl servers unaccessibility: https://github.com/dogecoin/dogecoin/pull/2399/checks

**Current behavior:**

```
Fetching openssl-1.0.1l.tar.gz from https://www.openssl.org/source/old/1.0.1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Couldn't connect to server
Fetching openssl-1.0.1l.tar.gz from https://bitcoincore.org/depends-sources
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
```
**Expected behavior**
Working build. If openssl down, backup link should work

**Steps to reproduce:**
Launch GH Action

**Comment by patricklodder** [comment]
Retriggered the Action for now. I'll push an openssl 1.0.1l mirror out there (and will go over all the other deps that may not have a tarball at bitcoincore.org anymore) later this week.

**Comment by ReverseControl** [comment]
@patricklodder  Is the OpenSSL server accessible yet? If you tell me where to look for the dependencies, and what to do once I get the tarballs I can take care of this one.

**Comment by ReverseControl** [comment]
@patricklodder What is the status of this one?

**Comment by patricklodder** [comment]
Meta. Moved it out of the queue.

**Comment by patricklodder** [comment]
I have a concept for caching dependency sources on ipfs instead of centralized web servers, but this needs further testing and integration before I propose it. Will target 1.14.6.

**Comment by micaelmalta** [comment]
@patricklodder do you need any help on this one?

**Comment by patricklodder** [comment]
I have all dependency archives pinned on ipfs, but there are issues with https gateway reliability for the larger files. I'll put up a centralized mirror for now.

**Comment by micaelmalta** [comment]
> I have all dependency archives pinned on ipfs, but there are issues with https gateway reliability for the larger files. I'll put up a centralized mirror for now.

Why don't you create a github repo dedicated for vendors?

**Comment by patricklodder** [comment]
> Why don't you create a github repo dedicated for vendors?

Filesize limitations - would need LFS.

I'm synching my ipfs folders to a central object storage right now so that we have a working backup available until we get something better.

**Comment by patricklodder** [comment]
Fixed with #2842, #2843 for 1.14, and #2849 for 1.21.

For historical builds down to v1.14.0, the dependencies are hosted too there, with the exception of mac_alias 1.1.0 and clang/llvm 3.0.0 because these expect conflicting filenames. In case you're building an older version, setting `SDK_URL` and `FALLBACK_DOWNLOAD_PATH` env variables will override the default coded into the tagged source. 

Versions prior to 1.14.0 have severe link rot issues on their original sources, and dependency sources for these are currently not mirrored.
